Show all scheduled report emails

xbryanc opened the pull request:
Each user should be able to see all scheduled reports he/she is a part of (if he/she is an admin, able to see all in domain). This is in a clearly labeled table underneath the owned scheduled reports.
Is under a flag ("SHOW_ALL_SCHEDULED_REPORT_EMAILS)
Changed wording for clarity (since owner may not be the only one editing now)
If the user is an admin, allowed him/her to see all emails in current domain
Admin privileges are same as owner (except cannot move around included reports)
Non-admin privileges are only allowed to view report, cannot see others' email addresses
Only shows up to the first 10 unowned scheduled reports, can click on a link to show page with all loaded
@dannyroberts

xbryanc committed:
show scheduled reports for all users who are in the email list

Is under a flag ("SHOW_ALL_SCHEDULED_REPORT_EMAILS)
Changed wording for clarity (since owner may not be the only one editing now)
If the user is an admin, allowed him/her to see all emails in current domain
Admin privileges are same as owner (except cannot move around included reports)
Non-admin privileges are only allowed to view report, cannot see others' email addresses

xbryanc committed:
Fix coding style and merge conflicts

added semicolon and fixed toggle merge conflict

xbryanc committed:
Merge branch 'master' into showreportemails

xbryanc committed:
Add newline

sticklerbot

dannyroberts commented on corehq/apps/hqwebapp/static/hqwebapp/js/multiselect_utils.js:
```diff
@@ -45,8 +45,17 @@ hqDefine('hqwebapp/js/multiselect_utils', function () {
+        if (!isOwner) {
+            $('#' + multiselectId).hide().after(
+                $('#' + multiselectId).children().each(function (el) {
+                    $(el).text();
```
Does this do anything? It looks like it just evaluates to some string but doesn't do anything with that.

xbryanc replied:
It does display it on the page as line-separated plain text

dannyroberts replied:
Can you double-check that, and double-check that this code is actually what's doing that (by also commenting it out and seeing if it still works). I can't see how this would do that.

xbryanc committed:
Merge branch 'showreportemails' of https://github.com/xbryanc/commcare-hq into showreportemails

Updating local code with merge conflict resolution

xbryanc committed:
Make HTML conversion more explicit and less error-prone

Made code to display reports when non-owner admins edit the page much clearer

xbryanc commented on corehq/apps/reports/views.py:
```diff
@@ -1012,6 +1067,7 @@ def page_context(self):
             'is_configurable_map': is_configurable_map,
             'languages_map': languages_map,
             'languages_for_select': languages_for_select,
+            'is_owner': self.request.couch_user._id == self.owner_id,
```
This is a bug right now, should also include self.is_new (the way it is now, people cannot make new scheduled reports). This will be fixed in the next commit.

xbryanc committed:
Fix important bug and separate owned and unowned scheduled reports

Fixed bug where people could not choose which reports to put in a new scheduled report

xbryanc committed:
separate scheduled reports based on ownership and add pseudo-pagination

Reports scheduled by others are now in a separate, "pseudo-paginated" table
Default limit is set to 10 reports, user can click on a link to display page with all reports
This was decided since pagination wouldn't be super useful to begin with

xbryanc committed:
fix linter errors

added spacing in arithmetic and fixed max line length

xbryanc committed:
fix redundancy and organization of code

Made template html file for scheduled_reports and others_scheduled reports
This improved the previous code that had a nearly identical table copy&pasted

dannyroberts commented on corehq/apps/reports/templates/reports/partials/scheduled_reports_table.html:
```diff
@@ -0,0 +1,135 @@
+<table class="table table-striped table-bordered">
+    {% if not is_owner %}
+    <h3>Other Scheduled Reports (managed by others)</h3>
+    {% endif %}
```
I would move this heading out to the main template, right above the line where the `others_scheduled_reports` table is included, for a couple reasons:
- An `<h3>` tag shouldn't be placed inside a `<table>`, so this should be moved up to before the `<table>` tag.
- This text isn't really part of the table, and is more a part of the page that includes the table; thus I think the main template is the better place to put it.

dannyroberts commented on corehq/apps/reports/templates/reports/partials/scheduled_reports_table.html:
```diff
@@ -0,0 +1,135 @@
+    <h3>Other Scheduled Reports (managed by others)</h3>
+    {% endif %}
+    <thead>
+        <tr>
```
Can you put an extra column here, only for the `others_scheduled_reports` table, called "Owner" that displays the name of the owner of that report? I noticed while testing on staging that you still couldn't easily tell who created a given report (and thus who you can ask to edit it).

dannyroberts reviewed (commented):
Looking good. Made a couple suggestions

xbryanc committed:
add "owner" column to others' scheduled reports table and clean up code

now displays owners so that users know who to contact for help/questions
